src/box/memtx_tree.cc

```diff
@@ -1871,6 +1892,13 @@ struct tree_read_view_iterator {
 	struct memtx_tree_key_data<USE_HINT> key_data;
 	/** BPS tree iterator. */
 	memtx_tree_iterator_t<USE_HINT> tree_iterator;
+	/**
+	 * Data that was fetched last, needed to make iterators stable.
+	 * Contains NULL as pointer to tuple only if there was no data fetched.
+	 * Otherwise, tuple pointer is not NULL, even if iterator is
+	 * exhausted - pagination relies on it.
+	 */
+	struct memtx_tree_data<USE_HINT> last;
```

Comment:
There's no need to store the last tuple because read view iterators are stable by definition.